cmd/kn/main.go

```diff
@@ -19,12 +19,22 @@ import (
 	"os"
 
 	"github.com/knative/client/pkg/kn/core"
+	"github.com/spf13/viper"
 )
 
+func init() {
+	core.InitializeConfig()
+}
+
 func main() {
-	err := core.NewKnCommand().Execute()
+	defer cleanup()
+	err := core.NewDefaultKnCommand().Execute()
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
 }
+
+func cleanup() {
+	viper.WriteConfig()
```

Comment:
Does this mean that any update to the config will only be written when the CLI exits? It may not be a big deal but if there are two CLIs active at the same time  then 1) they could over-write each other and while this will always be true, the longer they wait to write it the bigger chance it'll happen (instead of read, update, write immediately), 2) if a CLI command takes a while to run and a 2nd CLI command starts it won't see the updated config until the first one finishes, which might seem odd to the user.
Net: can we write after each update instead?

Reply:
Not sure that’s the write thing. Though I’ll consider it. One could argue that writing immediately is also not good since the command executing has not completed so having intermediary side effect is odd

Reply:
the current code writes it out no matter what, error or success, due to the defer.

Reply:
yes, but that will happen when it exits (or fails) so no intermediary changes.